src/Logary/Scheduling.fs

```diff
@@ -4,74 +4,87 @@ module Logary.Internals.Scheduling
+module Cancellation =
+  let create () = { cancelled = IVar.Now.create () }
+
+  let isCancelled cancellation = job {
+    return! (IVar.read cancellation.cancelled ^->. true) <|> (Alt.always false)
+  }
+
+  let cancel cancellation = job {
+    do! IVar.fill cancellation.cancelled ()
+  }
+
 type ScheduleMsg =
-  | Schedule of (obj -> unit) * obj * Duration * Duration * CancellationTokenSource ReplyChannel
-  | ScheduleOnce of (obj -> unit) * obj * Duration * CancellationTokenSource ReplyChannel
+  | Schedule of (obj -> unit) * obj * Duration * Duration * Cancellation
+  | ScheduleOnce of (obj -> unit) * obj * Duration * Cancellation
 
 module private Impl =
 
-  let ms (d : Duration) =
+  let ms (d: Duration) =
     d.ToTimeSpan().TotalMilliseconds |> int
 
-  let scheduleOnce (delay : Duration) msg receiver (cts: CancellationTokenSource) = async {
-    do! Async.Sleep (ms delay)
-    if cts.IsCancellationRequested then
-      cts.Dispose ()
-    else
+  let scheduleOnce delay msg receiver cts = job {
+    do! timeOutMillis delay
+    let! cancelled = Cancellation.isCancelled cts
+
+    if not cancelled then
       receiver msg
-    }
+  }
+
+  let scheduleMany initialDelay msg receiver delayBetween cts = Job.delay <| fun () ->
+    let rec loop time cts = job {
+      do! timeOutMillis time
+      let! cancelled = Cancellation.isCancelled cts
 
-  let scheduleMany initialDelay msg receiver delayBetween cts =
-    let rec loop time (cts: CancellationTokenSource) = async {
-      do! Async.Sleep time
-      if cts.IsCancellationRequested then
-        cts.Dispose ()
-      else
+      if not cancelled then
         receiver msg
-      return! loop delayBetween cts
+        return! loop delayBetween cts
     }
+
     loop initialDelay cts
 
-  let loop (inbox : IActor<_>) =
-    let rec loop () = async {
-      let! msg, _ = inbox.Receive ()
-      let cts = new CancellationTokenSource()
+  let loop (ch : Ch<ScheduleMsg>) =
+    let rec loop () = job {
+      let! msg = Ch.take ch
       match msg with
-      | Schedule (receiver, msg : 'a, initialDelay, delayBetween, replyChan) ->
-        Async.StartImmediate (scheduleMany (ms initialDelay) msg receiver (ms delayBetween) cts)
-        replyChan.Reply cts
+      | Schedule (receiver, msg : 'a, initialDelay, delayBetween, cts) ->
+        do! Job.start (scheduleMany (ms initialDelay) msg receiver (ms delayBetween) cts)
         return! loop ()
-      | ScheduleOnce (receiver, msg:'a, delay, replyChan) ->
-        Async.StartImmediate (scheduleOnce delay msg receiver cts)
-        replyChan.Reply cts
+      | ScheduleOnce (receiver, msg : 'a, delay, cts) ->
+        do! Job.start (scheduleOnce (ms delay) msg receiver cts)
         return! loop ()
     }
+
     loop ()
 
-/// Create a new scheduler actor
+/// Creates a new scheduler job
 let create () =
-  Actor.spawn (Ns.create "scheduler") Impl.loop
+  let ch = Ch.Now.create ()
+  Job.Global.start (Impl.loop ch)
```

Comment:
I think this can be done better with `server` from Hopac, as it has less overhead (according to docs)